lib/Service/Mail/Mailer.hs

```diff
@@ -6,58 +6,89 @@ module Service.Mail.Mailer
-      mailMessage = SMTP.simpleMail addrFrom addrTo cc bcc subject [emailBody]
-  if mailConfig ^. enabled
-    then liftIO $ SMTP.sendMailWithLogin mailHost mailUsername mailPassword mailMessage
-    else return ()
-
 sendRegistrationConfirmationMail :: Email -> U.UUID -> String -> AppContextM ()
 sendRegistrationConfirmationMail email userId hash = do
   dswConfig <- asks _appContextConfig
   let clientAddress = dswConfig ^. clientConfig . address
       clientLink = clientAddress ++ "/signup-confirmation/" ++ U.toString userId ++ "/" ++ hash
       link = "<a href=\"" ++ clientLink ++ "\">here</a>"
-      subject = "Confirmation Email"
-      body = TL.pack $ "Hi! For account activation you have to click " ++ link ++ "! Elixir DSW Wizard Team"
-  createEmail email subject body
+      mailName = dswConfig ^. mail . name
+      subject = TL.pack $ mailName ++ ": Confirmation Email"
+      body = TL.pack $ "Hi! For account activation you have to click " ++ link ++ "! " ++ mailName ++ " Team"
+  sendEmail email subject body
 
 sendRegistrationCreatedAnalyticsMail :: String -> String -> Email -> AppContextM ()
 sendRegistrationCreatedAnalyticsMail uName uSurname uEmail = do
   dswConfig <- asks _appContextConfig
   let analyticsAddress = dswConfig ^. analytics . email
-      subject = "DSW Wizard: New user"
+      mailName = dswConfig ^. mail . name
+      subject = TL.pack $ mailName ++ ": New user"
       body =
-        TL.pack $ "Hi! We have a new user (" ++ uName ++ " " ++ uSurname ++ ", " ++ uEmail ++
-        ") in our Wizard! Elixir DSW Wizard Team"
-  createEmail analyticsAddress subject body
+        TL.pack $ "Hi! We have a new user (" ++ uName ++ " " ++ uSurname ++ ", " ++ uEmail ++ ") in our Wizard! " ++
+        mailName ++
+        " Team"
+  sendEmail analyticsAddress subject body
 
 sendResetPasswordMail :: Email -> U.UUID -> String -> AppContextM ()
 sendResetPasswordMail email userId hash = do
   dswConfig <- asks _appContextConfig
   let clientAddress = dswConfig ^. clientConfig . address
       clientLink = clientAddress ++ "/forgotten-password/" ++ U.toString userId ++ "/" ++ hash
       link = "<a href=\"" ++ clientLink ++ "\">here</a>"
-      subject = "Reset Password"
-      body = TL.pack $ "Hi! You can set up a new password " ++ link ++ "! Elixir DSW Wizard Team"
-  createEmail email subject body
+      mailName = dswConfig ^. mail . name
+      subject = TL.pack $ mailName ++ ": Reset Password"
+      body = TL.pack $ "Hi! You can set up a new password " ++ link ++ "! " ++ mailName ++ " Team"
+  sendEmail email subject body
+
+-- --------------------------------
+-- PRIVATE
+-- --------------------------------
+createEmail to from subject plainBody "" _ = MIME.simpleMail' to from subject plainBody
+createEmail to from subject plainBody htmlBody attachments =
+  MIME.simpleMailInMemory to from subject plainBody htmlBody attachments
+
+makeConnection False host Nothing = SMTP.doSMTP host
```

Comment:
If it is not so hard, I would add here type specification

Reply:
As I remember, I think here is a problem with type specification that I would have to create own typeclass for that... I will try.

Reply:
Done in https://github.com/ds-wizard/dsw-server/pull/42/commits/69bc28099e3a562cff14378da6cb796e4b687ce6 (then this function removed in https://github.com/ds-wizard/dsw-server/pull/42/commits/b29cff1403d1834a815bc90290f3e23ed14c8846)